core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -184,8 +187,8 @@ public void onSettingsAckRead(ChannelHandlerContext ctx) {}
     public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int padding,
                               boolean endOfStream) throws Http2Exception {
         keepAliveChannelRead();
-        final HttpResponseWrapper res = getResponse(streamIdToId(streamId), endOfStream);
-        if (res == null) {
+        final HttpResponseWrapper res = getResponse(streamIdToId(streamId));
```

Comment:
nit: `getResponse(int id, boolean remove)` is used only in `onStreamClosed()`.
Could we remove `getResponse(int id, boolean remove)` and use `removeResponse()` instread?